### DISCUSSION TITLE
Support for TMWF02 Wi-Fi Fan Module

### DISCUSSION THREAD

**Discussion by joerg900** [arendst/Tasmota#14498] [Q&A] [answered]
Hello everyone,

I stumbled accross this [Tuya Wifi fan module](https://expo.tuya.com/product/842480) which should in theory allow a fan from 0 to 100% speed.

Has anyone heard about it or is possibly using it (with tasmota obviously)?  I haven't found any template or else so far and I don't know whether a it uses a compatible ESP. But it being tuya I tought there might be a chance.

Nonetheless I ordered one and will open it up as soon as it arrives. So I can at least have a look at the HW side.

Regards Joerg900

**Comment by sfromis** [comment, marked as answer]
By now, it would be rather surprising for a new Tuya design to include a "Tasmota-compatible" ESP8266 (or even ESP32) MCU. Best chances may be if it is using an easily swapable Wifi modules, which may be pin-compatible with ESP12 or another modules.

**Comment by jamaroney** [comment]
I bought one a few weeks back. Easy to install, but lower speeds caused a noticeable hum from the motor (my fan is a Hunter, so motor quality isn't the issue).

**Comment by digiblur** [comment]
Probably doing PWM on it too, so definitely don't use it on a fan that requires capacitors to change speed.

**Comment by jamaroney** [comment]
The Hunter fans definitely use capacitors to change speed, which explains the noise. I removed it right away. Thanks.

**Comment by remlei** [comment]
I actually use this one but not as a fan speed control but as a light dimmer works well, me hoping that this gets some support with esphome (given that the beken chip inside can now be flashed with esphome at least unofficially as of this post).

the voltage adjustment can be done from 1volt to 240-ish volt (242v direct from wall with a volt meter).
